packages/node/src/subcommands/testing.service.ts

```diff
@@ -0,0 +1,161 @@
+}
+
+@Injectable()
+export class TestingService {
+  constructor(
+    private readonly sequelize: Sequelize,
+    private readonly nodeConfig: NodeConfig,
+    private readonly storeService: StoreService,
+    @Inject('ISubqueryProject') private project: SubqueryProject,
+    private readonly apiService: ApiService,
+    private readonly indexerManager: IndexerManager,
+  ) {}
+
+  // eslint-disable-next-line @typescript-eslint/require-await
+  async init() {
+    const projectPath = this.project.root;
+    // find all paths to test files
+    const testFiles = this.findAllTestFiles(path.join(projectPath, 'src/test'));
+    // import all test files
+
+    const { compilerOptions } = require(path.join(
+      projectPath,
+      'tsconfig.json',
+    ));
+
+    const options: RegisterOptions = {
+      compilerOptions,
+      files: true,
+    };
+
+    register(options);
+
+    const tests = testFiles.map((file) => require(file));
+
+    logger.info(`Found ${tests.length} tests`);
+  }
+
+  async run() {
+    if ((global as any).subqlTests) {
+      await Promise.all(
+        (global as any).subqlTests.map(async (test) => {
+          await this.runTest(test);
+        }),
+      );
+    }
+  }
+
+  private findAllTestFiles(dirPath: string): string[] {
+    const files: string[] = [];
+
+    readdirSync(dirPath).forEach((file) => {
+      const filePath = path.join(dirPath, file);
+      const stat = statSync(filePath);
+
+      if (stat.isDirectory()) {
+        files.push(...this.findAllTestFiles(filePath));
+      } else if (filePath.endsWith('.test.ts')) {
+        files.push(filePath);
+      }
+    });
+
+    return files;
+  }
+
+  private async runTest(test: {
+    name: string;
+    blockHeight: number;
+    dependentEntities: Entity[];
+    expectedEntities: Entity[];
+    handler: HandlerFunction;
+    handlerKind: SubstrateHandlerKind;
+  }) {
+    // Fetch block
+    //TODO: this should be used made general for all networks.
+    // create a mapping to get the right fetch function based on the network?
+    const [block] = await SubstrateUtil.fetchBlocksBatches(
+      this.apiService.getApi(),
+      [test.blockHeight],
+    );
+
+    // Check filters match
+
+    // Init db
+    const modelRelations = getAllEntitiesRelations(this.project.schema);
+    for (let i = 0; i < modelRelations.models.length; i++) {
+      modelRelations.models[i].name = `${modelRelations.models[i].name}`;
+    }
+    await this.storeService.init(modelRelations, this.nodeConfig.subqueryName);
+
+    const store = this.storeService.getStore();
+    (global as any).store = store;
+
+    // Init entities
+    test.dependentEntities.map(async (entity) => {
+      const attrs = entity as unknown as CreationAttributes<Model>;
+      logger.info(JSON.stringify(attrs));
+      await (global as any).store.set(`${entity.name}`, entity.id, entity);
+    });
+
+    // Run handler
+    switch (test.handlerKind) {
```

Comment:
We should use existing classes where possible. Here we can use `IndexerManager.indexBlock`